src/plugins/dashboard/public/application/utils/get_nav_actions.tsx

```diff
@@ -313,18 +317,20 @@ export const getNavActions = (
         : DashboardConstants.CREATE_NEW_DASHBOARD_URL;
       history.push(pathname);
 
+      const newStateContainer: { [key: string]: any } = {};
       // This is only necessary for new dashboards, which will default to Edit mode.
-      stateContainer.transitions.set('viewMode', ViewMode.VIEW);
+      newStateContainer.viewMode = ViewMode.VIEW;
 
       // We need to reset the app state to its original state
       if (dashboard.panels) {
-        stateContainer.transitions.set('panels', dashboard.panels);
+        newStateContainer.panels = dashboard.panels;
       }
 
-      stateContainer.transitions.set('filters', dashboard.filters);
-      stateContainer.transitions.set('query', dashboard.query);
-      stateContainer.transitions.setOption('hidePanelTitles', dashboard.options.hidePanelTitles);
-      stateContainer.transitions.setOption('useMargins', dashboard.options.useMargins);
+      newStateContainer.filters = dashboard.filters;
+      newStateContainer.query = dashboard.query;
+      stateContainer.transitions.setDashboard(newStateContainer);
+      newStateContainer.transitions.setOption('hidePanelTitles', dashboard.options.hidePanelTitles);
```

Comment:
shouldn't these two lines still be calling `stateContainer` instead of `newStateContainer` for setting the options?

Reply:
Good catch. Addressed.